core/src/main/scala/org/bykn/bosatsu/rankn/Infer.scala

```diff
@@ -1046,10 +1013,7 @@ object Infer {
         expTy = rho.getType
         expTyRho <- assertRho(expTy, s"must be rho since $rho is a TypedExpr.Rho")
         envTys <- unifySelf(expTyRho)
-        envTypeVars <- getMetaTyVars(envTys.values.toList)
-        resTypeVars <- getMetaTyVars(expTyRho :: Nil)
-        forAllTvs = resTypeVars -- envTypeVars
-        q <- quantify(forAllTvs.toList, rho)
+        q <- TypedExpr.quantify(envTys, rho, zonk(_), { (m, n) => writeMeta(m, Type.TyVar(n)) })
```

Comment:
We could run this in the ref monad and lift it when done. It might be a bit faster.